asciidoctorj-core/src/main/java/org/asciidoctor/ast/AbstractNodeImpl.java

```diff
@@ -54,157 +63,211 @@ public DocumentRuby document() {
 
     @Override
     public DocumentRuby getDocument() {
-        return this.abstractNode.getDocument();
+        return (DocumentRuby) NodeConverter.createASTNode(getRubyProperty("document"));
+    }
+
+    public IRubyObject getRubyObject() {
+        return rubyNode;
```

Comment:
I think getRubyObject should be either protected or package-private. I'm not sure we should have it exposed (yet) as a public API method.